### PR TITLE
[MINOR][CORE] Make EventLogger codec be consistent between EventLogFileWriter and SparkContext

### PR DESCRIPTION
**Pull request by Ngone51**
### What changes were proposed in this pull request?

 `SparkContext` should use `EVENT_LOG_COMPRESSION_CODEC` to get the right codec name for EventLogger.

### Why are the changes needed?

#24921 added a new conf for EventLogger's compression codec. We should reflect this change into `SparkContext` as well. Though I didn't find any places that `SparkContext.eventLogCodec` really takes an effect, I think it'd be better to have it as a right value.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Pass Jenkins.

### CODE REVIEW

**Comment by Ngone51** [comment]
cc @cloud-fan @dongjoon-hyun

File: core/src/main/scala/org/apache/spark/SparkContext.scala [hunk 1]
```diff
@@ -444,7 +445,7 @@ class SparkContext(config: SparkConf) extends Logging {
     _eventLogCodec = {
       val compress = _conf.get(EVENT_LOG_COMPRESS)
       if (compress && isEventLogEnabled) {
-        Some(CompressionCodec.getCodecName(_conf)).map(CompressionCodec.getShortName)
+        Some(EventLogFileWriter.codecName(_conf)).map(CompressionCodec.getShortName)
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/SparkContext.scala]: Let's not make SparkContext further depend on these classes. Just change the other one to match how this works?

Comment (Ngone51) [reply core/src/main/scala/org/apache/spark/SparkContext.scala]: Any reason for reducing dependencies on "these classes" ?

"change the other one" ?
Maybe, just use `_conf .get(EVENT_LOG_COMPRESSION_CODEC)` without adding dependency on `EventLogFileWriter` should be OK ?

Comment (cloud-fan) [reply core/src/main/scala/org/apache/spark/SparkContext.scala]: yea `_conf.get(EVENT_LOG_COMPRESSION_CODEC)` should be good enough

Comment (dongjoon-hyun) [reply core/src/main/scala/org/apache/spark/SparkContext.scala]: +1 for @srowen and @cloud-fan 's advice.

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by Ngone51** [comment]
thanks! @cloud-fan @dongjoon-hyun @srowen

**Pull request closed without merging**